x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/response_actions_list.tsx

```diff
@@ -98,465 +98,482 @@ const StyledEuiCodeBlock = euiStyled(EuiCodeBlock).attrs({
-    ];
-    // filter out the host column
-    if (hideHostColumn) {
-      return columns.filter((column) => column.field !== 'agents');
-    }
-    return columns;
-  }, [agentIds, getTestId, itemIdToExpandedRowMap, onClickCallback]);
-
-  // table pagination
-  const tablePagination = useMemo(() => {
-    return {
-      // this controls the table UI page
-      // to match 0-based table paging
-      pageIndex: (queryParams.page || 1) - 1,
-      pageSize: queryParams.pageSize || 10,
-      totalItemCount,
-      pageSizeOptions: MANAGEMENT_PAGE_SIZE_OPTIONS as number[],
-    };
-  }, [queryParams, totalItemCount]);
-
-  // handle onChange
-  const handleTableOnChange = useCallback(
-    ({ page: _page }: CriteriaWithPagination<ActionDetails>) => {
-      // table paging is 0 based
-      const { index, size } = _page;
-      setQueryParams((prevState) => ({
-        ...prevState,
-        // adjust the page to conform to
-        // 1-based API page
-        page: index + 1,
-        pageSize: size,
-      }));
-      reFetchEndpointActionList();
-    },
-    [reFetchEndpointActionList, setQueryParams]
-  );
-
-  // compute record ranges
-  const pagedResultsCount = useMemo(() => {
-    const page = queryParams.page ?? 1;
-    const perPage = queryParams?.pageSize ?? 10;
-
-    const totalPages = Math.ceil(totalItemCount / perPage);
-    const fromCount = perPage * page - perPage + 1;
-    const toCount =
-      page === totalPages || totalPages === 1 ? totalItemCount : fromCount + perPage - 1;
-    return { fromCount, toCount };
-  }, [queryParams.page, queryParams.pageSize, totalItemCount]);
-
-  // create range label to display
-  const recordRangeLabel = useMemo(
-    () => (
-      <EuiText color="subdued" size="xs" data-test-subj={getTestId('endpointListTableTotal')}>
-        <FormattedMessage
-          id="xpack.securitySolution.responseActionsList.list.recordRange"
-          defaultMessage="Showing {range} of {total} {recordsLabel}"
-          values={{
-            range: (
-              <strong>
-                <EuiI18nNumber value={pagedResultsCount.fromCount} />
-                {'-'}
-                <EuiI18nNumber value={pagedResultsCount.toCount} />
-              </strong>
-            ),
-            total: <EuiI18nNumber value={totalItemCount} />,
-            recordsLabel: <strong>{UX_MESSAGES.recordsLabel(totalItemCount)}</strong>,
-          }}
+      [reFetchEndpointActionList, setQueryParams]
+    );
+
+    // compute record ranges
+    const pagedResultsCount = useMemo(() => {
+      const page = queryParams.page ?? 1;
+      const perPage = queryParams?.pageSize ?? 10;
+
+      const totalPages = Math.ceil(totalItemCount / perPage);
+      const fromCount = perPage * page - perPage + 1;
+      const toCount =
+        page === totalPages || totalPages === 1 ? totalItemCount : fromCount + perPage - 1;
+      return { fromCount, toCount };
+    }, [queryParams.page, queryParams.pageSize, totalItemCount]);
+
```

Comment:
I'm having a hard time understanding this code here. What is it trying to do? and is this calculation being done based on URL param values?

Reply:
just want to understand: is this being calculated from the API response data? 

It should not be calculated from URL params, because those can be manipulated.

Reply:
Not with URL prarms yet, this is being computed with the API results. We should do this with URL params when it has its own page.